lib/min_heap.rb

```diff
@@ -44,28 +52,60 @@ def to_s
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(1)
+  # Space complexity: O(1)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    return @store.length == 0
   end
 
   private
 
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(1)
+  # Space complexity: O(1)
   def heap_up(index)
-    
+    return if index == 0
+
+    parent_index = (index - 1)/2
+
+    if @store[index].key < @store[parent_index].key
+      swap(index, parent_index)      
+      heap_up(parent_index)
+    end
+
+    return
   end
 
   # This helper method takes an index and 
   #  moves it up the heap if it's smaller
   #  than it's parent node.
   def heap_down(index)
```

Comment:
No time/space complexities listed.  Since this is a recursive method it would give `remove` a space complexity of O(log n).